spec/integration/location_rerun_spec.rb

```diff
@@ -0,0 +1,133 @@
+require 'support/aruba_support'
+
+RSpec.describe 'Failed spec rerun location' do
+  include RSpecHelpers
+  include_context "aruba support"
+
+  before do
+    setup_aruba
+
+    # Setup some shared examples and call them in a separate file
+    # from where they are called to demonstrate how nested example ids work
+    write_file "some_examples.rb", "
+      RSpec.shared_examples_for 'a failing spec' do
+        it 'fails' do
+            expect(1).to eq(2)
+        end
+
+        context 'when you reverse it' do
+          it 'still fails' do
+            expect(2).to eq(1)
+          end
+        end
+    end
+    "
+
+    file = cd('.') { "#{Dir.pwd}/some_examples.rb" }
+    load file
+
+    write_file "non_local_shared_examples_spec.rb", "
+    RSpec.describe do
+        context 'the first context' do
+            it_behaves_like 'a failing spec'
+        end
+
+        context 'the second context' do
+            it_behaves_like 'a failing spec'
+        end
+    end
+"
+    # Setup some shared examples in the same file as where they are called
+    write_file "local_shared_examples_spec.rb", "
+      RSpec.describe do
+          shared_examples_for 'a failing spec' do
+              it 'fails' do
+                  expect(1).to eq(2)
+              end
+
+              context 'when you reverse it' do
+                it 'still fails' do
+                  expect(2).to eq(1)
+                end
+              end
+          end
+
+          context 'the first context' do
+              it_behaves_like 'a failing spec'
+          end
+
+          context 'the second context' do
+              it_behaves_like 'a failing spec'
+          end
+      end
+    "
```

Comment:
Can you convert the files to use the same unident trick, should neatedn them up to as theres a stray miasligned `"`

Reply:
Fixed the indents manually @JonRowe and have used `write_file_formatted` which calls `unindent` just to use the existing util 👌 

https://github.com/kykyi/rspec-core/blob/39f4fdfeea8ff6c70314ba385a47f20fc370cd2c/spec/support/aruba_support.rb#L72